tasks/__init__.py

```diff
@@ -66,7 +66,12 @@ def server(host=None, port=5000, debug=True, live=False, gitlogs=False):
         server.watch(os.path.join(HERE, 'website', 'static', 'public'))
         server.serve(port=port)
     else:
-        app.run(host=host, port=port, debug=debug, threaded=debug, extra_files=[settings.ASSET_HASH_PATH])
+        if settings.SECURE_MODE:
+            context = (settings.OSF_SERVER_CERT, settings.OSF_SERVER_KEY)
+        else:
+            context = None
+        app.run(host=host, port=port, debug=debug, threaded=debug, extra_files=[settings.ASSET_HASH_PATH], ssl_context=context)
```

Comment:
Can you clarify why you need a pre-made SSL context for this to work? (eg, is this a requirement because of the different servers working together?)

I notice that flask's `app.run` has an "ad-hoc" mode to generate this on the fly, and am wondering why we chose not to go that route. (it seems like it wuld save a step for anyone using this feature)

Adhoc mode is per the documentation of the thing flask's `app.run` is based on:
http://werkzeug.pocoo.org/docs/0.11/serving/#werkzeug.serving.run_simple
> ssl_context – an SSL context for the connection. Either an ssl.SSLContext, a tuple in the form (cert_file, pkey_file), the string 'adhoc' if the server should automatically create one, or None to disable SSL (which is the default).